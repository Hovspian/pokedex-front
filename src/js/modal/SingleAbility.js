import React from 'react';
import PropTypes from 'prop-types';
import { Popover, PopoverBody } from 'reactstrap';

import '../../styles/modal/SingleAbility.css';

const propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
};

class SingleAbility extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    }
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  render() {
    return (
      <span className="ability">
        <span id={`ability-${this.props.index}`}
              onMouseOver={this.toggle}
              onMouseOut={this.toggle}
        >
          {this.props.name}
        </span>
        <Popover placement="top"
                 isOpen={this.state.isOpen}
                 target={`ability-${this.props.index}`}
                 hideArrow={true}
        >
          <PopoverBody><h5>{this.props.description}</h5></PopoverBody>
        </Popover>
      </span>
    );
  }
}

SingleAbility.propTypes = propTypes;

export default SingleAbility;
