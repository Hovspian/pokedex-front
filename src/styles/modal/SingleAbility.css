.ability {
  position: relative;
  cursor: pointer;
  text-decoration: underline;
}

.ability .ability-tooltip {
  visibility: hidden;
  padding: 20px;
  border-radius: 10px;
  background-color: grey;
  position: absolute;
  bottom: 30px;
  z-index: 1;
  display: table-cell;
  vertical-align: center;
  width: 250px;
  max-width: 200%;
  margin-left: 20%;
  color: white;
}

.ability:hover .ability-tooltip {
  visibility: visible;
}
