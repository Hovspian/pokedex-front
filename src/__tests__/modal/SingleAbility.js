import React from 'react';
import { shallow } from 'enzyme';

import SingleAbility from '../../js/modal/SingleAbility';

describe('SingleAbility Component', () => {
  let props;
  beforeEach(() => {
    props = {
      name: "Thick Fat",
      description: "Halves damage from fire and ice moves.",
      index: 0,
    };
  });

  it('should render SingleAbility component without crashing', () => {
    shallow(<SingleAbility {...props} />);
  });

  it('should match SingleAbility component snapshot', () => {
    const wrapper = shallow(<SingleAbility {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
