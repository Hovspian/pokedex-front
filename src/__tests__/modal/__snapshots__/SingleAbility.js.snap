// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleAbility Component should match SingleAbility component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleAbility
    description="Halves damage from fire and ice moves."
    index={0}
    name="Thick Fat"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          id="ability-0"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
>
          Thick Fat
</span>,
        <Popover
          delay={
                    Object {
                              "hide": 0,
                              "show": 0,
                            }
          }
          hideArrow={true}
          isOpen={false}
          placement="top"
          placementPrefix="bs-popover"
          target="ability-0"
          toggle={[Function]}
>
          <PopoverBody
                    tag="div"
          >
                    <h5>
                              Halves damage from fire and ice moves.
                    </h5>
          </PopoverBody>
</Popover>,
      ],
      "className": "ability",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Thick Fat",
          "id": "ability-0",
          "onMouseOut": [Function],
          "onMouseOver": [Function],
        },
        "ref": null,
        "rendered": "Thick Fat",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <PopoverBody
            tag="div"
>
            <h5>
                        Halves damage from fire and ice moves.
            </h5>
</PopoverBody>,
          "delay": Object {
            "hide": 0,
            "show": 0,
          },
          "hideArrow": true,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-popover",
          "target": "ability-0",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h5>
              Halves damage from fire and ice moves.
</h5>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Halves damage from fire and ice moves.",
            },
            "ref": null,
            "rendered": "Halves damage from fire and ice moves.",
            "type": "h5",
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            id="ability-0"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
>
            Thick Fat
</span>,
          <Popover
            delay={
                        Object {
                                    "hide": 0,
                                    "show": 0,
                                  }
            }
            hideArrow={true}
            isOpen={false}
            placement="top"
            placementPrefix="bs-popover"
            target="ability-0"
            toggle={[Function]}
>
            <PopoverBody
                        tag="div"
            >
                        <h5>
                                    Halves damage from fire and ice moves.
                        </h5>
            </PopoverBody>
</Popover>,
        ],
        "className": "ability",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Thick Fat",
            "id": "ability-0",
            "onMouseOut": [Function],
            "onMouseOver": [Function],
          },
          "ref": null,
          "rendered": "Thick Fat",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <PopoverBody
              tag="div"
>
              <h5>
                            Halves damage from fire and ice moves.
              </h5>
</PopoverBody>,
            "delay": Object {
              "hide": 0,
              "show": 0,
            },
            "hideArrow": true,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-popover",
            "target": "ability-0",
            "toggle": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h5>
                Halves damage from fire and ice moves.
</h5>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Halves damage from fire and ice moves.",
              },
              "ref": null,
              "rendered": "Halves damage from fire and ice moves.",
              "type": "h5",
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
